controller/api/models.py

```diff
@@ -381,7 +381,7 @@ def _healthcheck(self, containers, config):
         # HACK (bacongobbler): we need to wait until publisher has a chance to publish each
         # service to etcd, which can take up to 20 seconds.
         time.sleep(20)
-        for i in range(0, 4):
+        for i in range(len(intervals)):
```

Comment:
`range` creates a copy of the list in-memory to be evaluated on the first iteration, so it won't be continuously evaluated. This also includes any modifications done to the list inside the loop, which will not be evaluated. :)

You do bring up a good point though. I know @mboersma prefers `xrange` since we're on 2.7. `xrange` will avoid creating a copy of the list by lazily evaluating the list, acting as a generator. I'll change `range` to `xrange` for consistency with the rest of the code.